firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -612,47 +648,49 @@ private static RuntimeException deserializeError(ErrorPath path, String reason)
       do {
         // Add any setters
         for (Method method : currentClass.getDeclaredMethods()) {
-          if (shouldIncludeSetter(method)) {
-            String propertyName = propertyName(method);
-            String existingPropertyName = properties.get(propertyName.toLowerCase(Locale.US));
-            if (existingPropertyName != null) {
-              if (!existingPropertyName.equals(propertyName)) {
-                throw new RuntimeException(
-                    "Found setter on "
-                        + currentClass.getName()
-                        + " with invalid case-sensitive name: "
-                        + method.getName());
-              } else {
-                Method existingSetter = setters.get(propertyName);
-                if (existingSetter == null) {
-                  method.setAccessible(true);
-                  setters.put(propertyName, method);
-                  applySetterAnnotations(method);
-                } else if (!isSetterOverride(method, existingSetter)) {
-                  // We require that setters with conflicting property names are
-                  // overrides from a base class
-                  if (currentClass == clazz) {
-                    // TODO: Should we support overloads?
-                    throw new RuntimeException(
-                        "Class "
-                            + clazz.getName()
-                            + " has multiple setter overloads with name "
-                            + method.getName());
-                  } else {
-                    throw new RuntimeException(
-                        "Found conflicting setters "
-                            + "with name: "
-                            + method.getName()
-                            + " (conflicts with "
-                            + existingSetter.getName()
-                            + " defined on "
-                            + existingSetter.getDeclaringClass().getName()
-                            + ")");
-                  }
-                }
-              }
+          if (!shouldIncludeSetter(method)) {
```

Comment:
To reviewer: no behavior change here (supposedly), only refactor to reduce number of nested if/else.